Compiler/AST/ASTType.cs

```diff
@@ -120,7 +120,7 @@ public static (List<ASTError>, ASTType) Parse(IParser parser, List<ASTAnnotation
 
         }
 
-        public object Clone()
+        public ASTType Clone()
```

Comment:
Why not use the IClonable function?

Reply:
Was implemented as a simple Clone. Still need to refactor that to the IClonable interface. Will be next time once we've revisited the cloning.